### ISSUE TITLE
pull stops when stream goes away

### ISSUE THREAD

**Issue by winkmichael** [sergey-dryabzhinsky/nginx-rtmp-module#221]
Hello Sergey,

We've noticed that the pull command doesn't appear to retry forever. For example if you start nginx and the stream is unavailable on the first try, pull never tries again.  I see the pull_reconnect code, but can't figure why it doesn't seem to actaully reconnect.

The second issue is if you start nginx and the stream you are pulling is available and goes away, pull never seems to try again.

Perhaps could  be 2 parameters ideally, related to the pull_reconnect code. Retry interval, and Retry duration 0 being forever.

Let me know if you'd like me to setup a demo for you!

Thanks for all your hard work!

**Comment by tixoi** [comment]
Hello and sorry if I intervene with my remarks.
As far as I know, at least in the original arut's fork, _pull_ is of two very different types - static and dynamic (by default).
In case you _pull_ directive is with _static_, it will be reconnecting continuously with timeout _pull_reconnect_.
In case you _pull_ directive is without _static_, pulling will be closed every time you pull source (publisher) goes away or the last client connected to the stream will be disconnected.
Looks like this was the initial arut's idea for the pull - use static if you need it all the time, or use dynamic and it will be triggered by the connecting clients and by the source (publisher).

**Comment by winkmichael** [comment]
Correct, I have the static flag in there. However if the stream is unavailable when nginx starts it will never pull again, and secondly it appears if the stream goes away for more than a few minutes it will never pull agan either.

**Comment by winkmichael** [comment]
I should also note i've tried variations of the idle_publisher with no success.

**Comment by tixoi** [comment]
I am sorry, but I can not confirm this issue.
For me static pull reconnects to the source every time it is available.
You should look at log files, stats page and try to find out what is the problem with your source.

**Comment by winkmichael** [comment]
I am not quite certain still doing more testing, but it appears if you are creating HLS & Dash in the application being pulled it seems to be the cause.

Will confirm today

**Comment by winkmichael** [comment]
application 3000pull {
        live on;
        deny publish all;
        allow play all;

        pull rtmp://potato.com/live/ name=brown static 'flashVer=WIN 14,0,0,176' live=1;
        pull rtmp://potato.com/live/ name=white static 'flashVer=WIN 14,0,0,176' live=1;
    
        hls on;
        dash on;
        hls_path /potatoes/3000/hls/;
        hls_playlist_length 9000ms;
        hls_fragment 3000ms;

        dash_path /potatoes/3000/dash/;
        dash_playlist_length 9000ms;
        dash_fragment 3000ms;
        }


confirmed issue occurs

**Comment by winkmichael** [comment]
Tried a build from today, same issue when HLS and Dash are enabled. Effectively we are pulling and converting to HLS an Dash
